src/engine/lib/GUIDOScoreMap.cpp

```diff
@@ -86,6 +86,12 @@ bool TimeSegment::include(const TimeSegment& ts) const
 	return (s1a <= s2a) && (s1b >= s2b);
 }
 
+bool TimeSegment::startEqual(const TimeSegment& ts) const
```

Comment:
you could miss dates that are equal but expressed using different fractions (e.g. 2/4 and 1/2). I suggest to use Fraction or to compare using a float reduction.